entities/Offer.php

```diff
@@ -174,16 +174,16 @@ public function update(){
 			return false;
 		}
 	}
-
 	public function delete(){
-		$stmt=readOne($this->id);
-		if($stmt->rowCount()==1){
-			$sql = "DELETE FROM offers WHERE id=:id";
-			$stmt= $this->connection->prepare($sql);
+        $stmt=$this->connection->prepare("select organization_id from offers where id=:id");
```

Comment:
You don't need to do this query. There was a readOne statement at line 179, so $this->organization_id will be populated by that query, and you can just check if it matches $_SESSION['organization_id']

Reply:
I have edited my code so that it uses the readOne function, and I have tested it to ensure that it works.